scripts/convert_merged_tract_to_dpdd.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+
+"""Take LSST DESC DC2 Generic Catalog Reader accessible Object table
+(merged_tract trim cat file) and produce output files labeled with DPDD names.
+
+Output in HDF, FITS, and Parquet
+
+For each format there is one file per tract+patch.
+
+TODO: Exploring writing HDF and Parquet files in full tracts
+because those file formats allow easy writing to existing files.
+Keep FITS files in tracdt+patch.
+
+Requires
+
+pandas  # version >= 0.21
+generic-catalog-reader
+LSSTDESC/gcr-catalog
+
+and either
+pyarrow or fastparquet.
+"""
+
+import sys
+
+from astropy.table import Table
+import pandas as pd
+
+import GCRCatalogs
+
+
+def convert_all_to_dpdd(reader='dc2_coadd_run1.1p', **kwargs):
+    """Produce DPDD output files for all available tracts in GCR 'reader'."""
+    trim_config = {'filename_pattern': 'trim_merged_tract_.*\.hdf5$'}
+    cat = GCRCatalogs.load_catalog(reader, trim_config)
+
+    convert_cat_to_dpdd(cat, **kwargs)
+
+
+def convert_tract_to_dpdd(tract, reader='dc2_coadd_run1.1p', **kwargs):
+    """Produce DPDD output files for specified 'tract' and GCR 'reader'."""
+    trim_thistract_config = {
+        'filename_pattern': 'trim_merged_tract_{:04d}\.hdf5$'.format(tract)}
+    cat = GCRCatalogs.load_catalog(reader, trim_thistract_config)
+
+    convert_cat_to_dpdd(cat, **kwargs)
+
+
+def convert_cat_to_dpdd(cat, reader='dc2_coadd_run1.1p',
+                        **kwargs):
+    """Save DPDD-named columns files for all tracts, patches from input GCR catalog.
+    """
+    columns = cat.list_all_quantities()
+    columns.extend(['tract', 'patch'])
+
+    quantities = cat.get_quantities(columns, return_iterator=True)
+    for quantities_this_patch in quantities:
+        write_quantities_to_files(quantities_this_patch, **kwargs)
```

Comment:
This is more a personal preference, but I'd convert the dict to pandas dataframe here.
```python
for quantities_this_patch in quantities:
    quantities_this_patch = pd.DataFrame.from_dict(quantities_this_patch)
    write_quantities_to_files(quantities_this_patch, **kwargs)
```
And let `write_quantities_to_files` takes dataframe directly. 

This way, `quantities_this_patch` won't take up twice the memory.